broadinstitute/single_cell_portal_core
# Refine gene search autocomplete in Study page (SCP-3432)

This fixes a UI/UX bug with gene search autocomplete suggestion, and refines its other aspects in the Study page.  It shows:

* Exact matches first
* Suggestions (performantly!) even upon entering 1 character
* Prefix matches above other substring matches
* Similar results even if they're not precise substring matches

Old UI/UX bug: exact match not visible by default
<img width="672" alt="Screen Shot 2021-06-23 at 12 57 06 PM" src="https://user-images.githubusercontent.com/1334561/123138130-b3799b80-d422-11eb-8ebc-ce36452c47e5.png">

New UI/UX: exact match ranked first
<img width="678" alt="Screen Shot 2021-06-23 at 12 58 54 PM" src="https://user-images.githubusercontent.com/1334561/123138250-cd1ae300-d422-11eb-9a4c-15e90b22170b.png">

To test:
* Go to a Study page with many genes
* Search for "nf2"
* Verify that "Nf2" (or "NF2") appears first among autocomplete suggestions

This satisfies SCP-3432.
